apis/datadoghq/v2alpha1/utils.go

```diff
@@ -57,3 +58,13 @@ func ConvertCustomConfig(config *CustomConfig) *commonv1.CustomConfig {
 		ConfigMap:  configMap,
 	}
 }
+
+// IsHostNetworkEnabled returns whether the pod should use the host's network namespace
+func IsHostNetworkEnabled(dda *DatadogAgent) bool {
```

Comment:
for dogstatsd, we only care about the agent component right? instead of checking all components, should we just check the agent component?